staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -232,7 +234,18 @@ func (f *ConfigFlags) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, e
 	}
 
 	discoveryCacheDir := computeDiscoverCacheDir(filepath.Join(homedir.HomeDir(), ".kube", "cache", "discovery"), config.Host)
-	return diskcached.NewCachedDiscoveryClientForConfig(config, discoveryCacheDir, httpCacheDir, time.Duration(10*time.Minute))
+	var ttl time.Duration
```

Comment:
this can be refactored to:
```
ttl := time.Duration(10 * time.Minute)
if f.ClientCacheTTL != nil {
	cacheTTL, err := time.ParseDuration(*f.ClientCacheTTL)
	if err != nil {
		// return err
	}
	ttl = cacheTTL
}
```